src/ui/src/app/dashboard/dashboard-body/main/main-page/main-page.component.ts

```diff
@@ -33,35 +33,64 @@ export class MainPageComponent implements OnInit {
   currentDate = new Date();
   weekDays = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
   calendarDays: CalendarDay[] = [];
-  day= this.calendarDays;
-
-  todaysAbsences: Absence[] = [
-    { name: 'Eldo P Joy', department: 'Developer', reason: 'Sick Leave' },
-    { name: 'Aravind', department: 'Developer', reason: 'Vacation' },
-    { name: 'Siva Prakash', department: 'QA', reason: 'Personal Leave' },
-    { name: 'Irfan', department: 'Developer', reason: 'Work from Home' },
-    { name: 'Prem', department: 'Finance', reason: 'Sick Leave' }
-  ];
-   absence=this.todaysAbsences;
-
+  day = this.calendarDays;
   holidays: Holiday[] = [
     { date: '2024-01-26', name: 'Republic Day' },
     { date: '2024-08-15', name: 'Independence Day' },
     { date: '2024-10-02', name: 'Gandhi Jayanti' },
-    {date : '2024-11-15',name: 'National Holiday'},
+    { date: '2024-11-15', name: 'National Holiday' },
   ];
 
-  
+  //leave balance 
+  leaveManagementService = inject(LeaveManagementService)
+  userService = inject(WolfDenService)
+  destroyRef = inject(DestroyRef)
+  leaveList: ILeaveBalanceList[] = [];
+  todaysAbsences: ItodaysAbsence[]=[];
+  leaveType1: string = '';
+leaveBalance1: number = 0;
+
+leaveType2: string = '';
+leaveBalance2: number = 0;
+
+leaveType3: string = '';
+leaveBalance3: number = 0;
   ngOnInit() {
     this.generateCalendar();
+    //todays absence
+    this.userService.getTodaysAbsence().subscribe((data)=>{
+      this.todaysAbsences=data;
+    })
+
+    this.leaveManagementService.getLeaveBalance(this.userService.userId)
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe((data) => {
+        this.leaveList = data;
+        if (this.leaveList.length > 2) {
```

Comment:
what is the logic behind 2

Reply:
we have 3 cards on top, if there is no 3 type of leaves it will not display) displaying single cards is not good looking so only if we have 3 types of leave the card will display